azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/arm/collection/implementation/ResourcesImpl.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.resources.fluentcore.arm.collection.implementation;
+
+import java.io.IOException;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.management.resources.fluentcore.arm.ResourceUtils;
+import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsGettingByGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsGettingById;
+
+/**
+ * Base class for resource collection classes.
+ * (Internal use only)
+ * @param <T> the individual resource type returned
+ * @param <ImplT> the individual resource implementation
+ */
+public abstract class ResourcesImpl<T, ImplT>
+    implements
```

Comment:
great point, though  now in the process of addressing it I discovered that Deployments are weird - they're Groupable but they do not have a region, so they cannot inherit from GroupableResource right now. I'm looking into what should be one about that...